core/config/value.go

```diff
@@ -479,23 +480,34 @@ func (cv Value) getValueStruct(key string, target interface{}) (interface{}, boo
 
 					item := destSlice.Index(ai)
 					item.Set(reflect.ValueOf(itemValue))
-					found = true
 				} else {
 					break
 				}
 			}
 			if destSlice.Len() > 0 {
 				fieldValue.Set(destSlice)
 			}
-		}
-
-	}
+		case bucketMap:
+			destMap := reflect.MakeMap(fieldType).Interface()
+			val := global.GetValue(childKey).Value()
 
-	if found {
-		if errs := validator.Validate(target); errs != nil {
-			return nil, true, errs
+			if val != nil {
+				switch v := val.(type) {
+				case map[interface{}]interface{}:
```

Comment:
co case for `map[string]interface{}`? can you comment on why that is? also, a `switch` statement with a single `case` isn't very useful, perhaps this would be clearer as an if statement?